ci/format.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -euo pipefail
+
+ruff_fix=("--fix")
+if [[ ${1-} == "--check" ]]; then
+    ruff_fix=("--diff")
+    ruff_format=("--check" "--diff")
+    shift
+fi
+
+files=("$@")
+
+if [[ $# == 0 ]]; then
+    files=(".")
+fi
+
+EXIT_CODE=0
+
+poetry run ruff check --select I001,I002 "${ruff_fix[@]}" "${files[@]}" || EXIT_CODE=1
```

Comment:
If it makes sense to include/exclude specific formatting rules, let's set them in pyproject.yaml, not adhoc here